core/deployment/src/main/java/io/quarkus/deployment/dev/RuntimeUpdatesProcessor.java

```diff
@@ -666,4 +675,28 @@ public void close() throws IOException {
         FSWatchUtil.shutdown();
     }
 
+    private Map<Path, Long> expandGlobPattern(Path root, Path configFile) {
+        PathMatcher pathMatcher = FileSystems.getDefault().getPathMatcher("glob:" + configFile.toString());
+        Map<Path, Long> files = new HashMap<>();
+        try {
+            Files.walkFileTree(root, new SimpleFileVisitor<Path>() {
```

Comment:
I think we should only call this if an exists() check fails, so we don't have to walk the file system for no reason.

Reply:
Yes, this method is only called if the file does not exist.